src/ECS/Registry.h

```diff
@@ -147,11 +135,12 @@ class Registry
 	template <typename... Components>
 	[[nodiscard]] decltype(auto) Size() const
 	{
-		return _registry.view<Components...>().size();
+		return GetRegistry().view<Components...>().size();
 	}
 
 private:
-	entt::registry _registry;
+	[[nodiscard]] virtual entt::registry& GetRegistry() = 0;
```

Comment:
Moving to a virtual function adds an indirection and a guaranteed cache miss.
As mentioned in #626 it might not be necessary to make this virtual.

As I see it, this doesn't improve compilation time: There's only `_registry` that gets hidden and entt.hpp is still included in the interface.
This class, and `_registry especially`, is a hotspot for performance so I don't think making it an abstract base class is the right move.